SeaBattle.API/Services/EngineService.cs

```diff
@@ -0,0 +1,28 @@
+﻿using BoardNamespace;
+using EngineNamespace;
+using SeaBattle.API.Controllers;
+using SeaBattle.Engine.Interfaces;
+using SeaBattle.Values;
+using WebApp.Controllers;
+
+namespace SeaBattle.API.Services
+{
+    public class EngineService
+    {
+        public readonly Board board = new Board();
+        public readonly EngineNamespace.Engine engine;
+
+        public EngineService()
+        {
+            var outputHandler = new WebOutput();
+            engine = new EngineNamespace.Engine(board, null, outputHandler);
+            engine.Start();
+            engine.PlaceShips();
+        }
+
+        public void RestartGame()
+        {
+            engine.Restart();
+        }
```

Comment:
Issue: This looks really strange. You should never have some logic in the constructor. It may lead to unexpected behaviour. You should think about redesigning you domain part since right now it looks not really good.